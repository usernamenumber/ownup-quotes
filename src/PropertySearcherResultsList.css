.PropertySearcherResultsList {
    width: 700px;
    margin-left: 50px;
    margin-top: 30px;
    font-size: 80%;
}
.PropertySearcherResultsList table {
    text-align: left;
    border: thin solid lightgrey;
}
.PropertySearcherResultsList th {
    text-align: left;
    color: lightgrey;
    padding: 1em;
}
.PropertySearcherResultsList td {
    width: 200px;
    padding: 1em;
    border-top: thin solid lightgrey;
}

.PropertySearcherResultsList td.lenderName {
  /* white-space: nowrap; */
  overflow: hidden;
  text-overflow: ellipsis;
}

.PropertySearcherResultsList td.interestRate::after {
    content: '%';
}
.PropertySearcherResultsList td.apr::after {
    content: '%';
}
.PropertySearcherResultsList td.closingCosts::before {
    content: '$';
}
.PropertySearcherResultsList td.monthlyPayment::before {
    content: '$';
}