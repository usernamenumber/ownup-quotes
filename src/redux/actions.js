const prefix = 'PropertySearcher';
const actions = {
    updateQuotes: (quotes) => ({
        type: `${prefix}/updateQuotes`,
        payload: quotes,
    }),
    quoteUpdateStart: () => ({
        type: `${prefix}/quotesUpdateStart`,
        payload: true,
    }),
    quoteUpdateFinish: () => ({
        type: `${prefix}/quotesUpdateFinish`,
        payload: false,
    }) 
}

export default actions;