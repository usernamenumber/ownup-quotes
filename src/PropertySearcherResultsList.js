import './PropertySearcherResultsList.css';
import { connect } from 'react-redux'
import utils from './utils';

const PropertySearcherResultsList = (props) => {
    console.log("results rendering");
    console.log(props);
    const output_rows = props.quotes.map( (quotedata) => {
        const {lenderName, loanType, interestRate, closingCosts, monthlyPayment, apr} = quotedata;
        const key = `${lenderName}::${loanType}`;
        return (
            <tr key={key}>
                <td className='lenderName'>{lenderName}</td>
                <td className='loanType'>{loanType}</td>
                <td className='interestRate'>{Number.parseFloat(interestRate).toFixed(3)}</td>
                <td className='closingCosts'>{Number.parseFloat(closingCosts).toFixed(0)}</td>
                <td className='monthlyPayment'>{Number.parseFloat(monthlyPayment).toFixed(2)}</td>
                <td className='apr'>{Number.parseFloat(apr).toFixed(3)}</td>
            </tr>
        );
    }); 
    return (
        <div className="PropertySearcherResultsList">
            <table cellspacing="0">
                <thead>
                    <tr>
                        <th>Lender</th>
                        <th>Product</th>
                        <th>Rate</th>
                        <th>Closing Costs</th>
                        <th>Monthly Payment</th>
                        <th>APR</th>
                    </tr>
                </thead>
                <tbody>
                    {output_rows}
                </tbody>
            </table>
        </div>
    )
}

export default connect(
    (state) => ({
        quotes: state.quotes,
    }),
    null
)(PropertySearcherResultsList);