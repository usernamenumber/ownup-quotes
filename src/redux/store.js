import { createStore } from 'redux';

export const initial_store_data = {
    quotes: [],
}

const prefix = 'PropertySearcher';
export const store = createStore(
    (state = initial_store_data, action) => {
        switch(action.type) {
            case `${prefix}/updateQuotes`:
                return {...state, quotes: action.payload};
            case `${prefix}/quotesUpdateStart`:
                console.log('setting quotes_updating true');
                return {...state, quotes_updating: true};
            case `${prefix}/quotesUpdateFinish`:
                console.log('setting quotes_updating false');
                return {...state, quotes_updating: false};

            default: 
                if ( ! action.type.startsWith('@')) {
                    console.log(`prefix is ${prefix}`);
                    console.warn(`redux store got unrecognized action '${action.type}'`);
                }
                return state;
        }
    } ,
    window.__REDUX_DEVTOOLS_EXTENSION__ && window.__REDUX_DEVTOOLS_EXTENSION__(),
); 

export default store;