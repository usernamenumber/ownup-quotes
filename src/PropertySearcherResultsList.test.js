import Enzyme from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import PropertySearcherResultsList from './PropertySearcherResultsList';
import { Provider } from 'react-redux';
import configureStore from 'redux-mock-store';
import renderer from 'react-test-renderer';
import { initial_store_data } from "./redux/store";

Enzyme.configure({adapter: new Adapter() });

const quotes = [
    {
        "lenderName": "TFB Federal Credit Union",
        "loanType": "7/1 ARM",
        "interestRate": 3.5,
        "closingCosts": 2000,
        "monthlyPayment": 449.0446878088235,
        "apr": 3.660036729314016
    },
    {
        "lenderName": "Bates College Credit Union",
        "loanType": "10/1 ARM",
        "interestRate": 3.125,
        "closingCosts": 2570,
        "monthlyPayment": 4420.9621026749655,
        "apr": 3.396947112664067,
    }
]    

let store;
const mockStore = configureStore([]);
const setupComponent = () => {
store = mockStore({...initial_store_data, quotes});
  console.log(`store.getstate = ${store.getState()}`);
  console.log(store.getState());
  return (
    <Provider store={store}>
      <PropertySearcherResultsList />
    </Provider>
  )
}

it('renders correctly (matches snapshot)', () => {
    const tree = renderer
      .create(setupComponent())
      .toJSON();
    expect(tree).toMatchSnapshot();
});