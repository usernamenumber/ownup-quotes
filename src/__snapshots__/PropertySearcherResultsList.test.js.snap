// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly (matches snapshot) 1`] = `
<div
  className="PropertySearcherResultsList"
>
  
  <table
    cellSpacing="0"
  >
    <thead>
      <tr>
        <th>
          Lender
        </th>
        <th>
          Product
        </th>
        <th>
          Rate
        </th>
        <th>
          Closing Costs
        </th>
        <th>
          Monthly Payment
        </th>
        <th>
          APR
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td
          className="lenderName"
        >
          TFB Federal Credit Union
        </td>
        <td
          className="loanType"
        >
          7/1 ARM
        </td>
        <td
          className="interestRate"
        >
          3.500
        </td>
        <td
          className="closingCosts"
        >
          2000
        </td>
        <td
          className="monthlyPayment"
        >
          449.04
        </td>
        <td
          className="apr"
        >
          3.660
        </td>
      </tr>
      <tr>
        <td
          className="lenderName"
        >
          Bates College Credit Union
        </td>
        <td
          className="loanType"
        >
          10/1 ARM
        </td>
        <td
          className="interestRate"
        >
          3.125
        </td>
        <td
          className="closingCosts"
        >
          2570
        </td>
        <td
          className="monthlyPayment"
        >
          4420.96
        </td>
        <td
          className="apr"
        >
          3.397
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
